alf/environments/gym_wrappers.py

```diff
@@ -138,6 +117,51 @@ def transform_observation(self, observation, field=None):
         raise NotImplementedError("transform_observation is not implemented")
 
 
+@gin.configurable
+class ImageChannelFirst(BaseObservationWrapper):
+    """Make images in observations channel_first. """
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space, field=None):
+        if isinstance(observation_space, gym.spaces.Box):
+            if self._need_channel_transpose(observation_space.shape):
+                low = observation_space.low
+                high = observation_space.high
+                if np.isscalar(low) and np.isscalar(high):
+                    shape = observation_space.shape[::-1]
+                    return gym.spaces.Box(
+                        low=low,
+                        high=high,
+                        shape=shape,
+                        dtype=observation_space.dtype)
+                else:
+                    low = self._channel_first(
+                        observation_space.low, transpose=True)
+                    high = self._channel_first(
+                        observation_space.high, transpose=True)
+                    return gym.spaces.Box(
+                        low=low, high=high, dtype=observation_space.dtype)
+        return observation_space
+
+    def transform_observation(self, observation, field=None):
+        transpose = self._need_channel_transpose(observation.shape)
```

Comment:
`if self._need_channel_transpose(observation.shape):
    self._make_channel_first(observation)
`
This is more readable and has a cleaner function signature.

Reply:
updated